### PR TITLE
Tx data contains tx type

### PR DESCRIPTION
**Pull request by batconjurer**
We support various kinds of tx types: Raw, Decrypted, and Wrapper. Previously, we would have to try and deserialize a Tx's data into each of these different types to figure out which type it was. With this PR, we wrap all transactions in TxType enum and serialize that instead. This way, we can always deserialize and immediately know the type.

### CODE REVIEW

File: apps/src/lib/node/ledger/shell/finalize_block.rs [hunk 1]
```diff
@@ -56,9 +56,13 @@ where
             if ErrorCodes::from_u32(tx.result.code).unwrap()
                 == ErrorCodes::InvalidSig
             {
-                let wrapper =
-                    WrapperTx::try_from(&Tx::try_from(tx.tx.as_ref()).unwrap())
-                        .unwrap();
+                let wrapper = if let Ok(TxType::Wrapper(wrapper)) =
+                    TxType::try_from(Tx::try_from(tx.tx.as_ref()).unwrap())
```
Comment (tzemanovic) [inline apps/src/lib/node/ledger/shell/finalize_block.rs:60]: I'm not sure if this unwrap is safe, in `shell/process_proposal`, it looks like there are many cases where `TxResult` is converted to `Response::ProcessProposal` with default (empty) tx bytes

**Review by tzemanovic — APPROVED** [review]
besides the one comment, LGTM!

File: apps/src/lib/node/ledger/shell/finalize_block.rs (hunk 1 above)
Comment (batconjurer) [reply apps/src/lib/node/ledger/shell/finalize_block.rs:60]: The orignal tx is saved in the shim. FinalizeBlock sees the exact same tx as ProcessProposal

Comment (batconjurer) [reply apps/src/lib/node/ledger/shell/finalize_block.rs:60]: See abcipp_shim.rs lines 87-94

**Pull request closed without merging**